pkg/framework/context/context.go

```diff
@@ -158,16 +162,33 @@ func (p *Provider) tryToHandle(svc service.InboundHandler, msg service.DIDCommMs
 }
 
 // InboundMessageHandler return an inbound message handler.
-func (p *Provider) InboundMessageHandler() transport.InboundMessageHandler {
-	return func(message []byte, myDID, theirDID string) error {
-		msg, err := service.ParseDIDCommMsgMap(message)
+func (p *Provider) InboundMessageHandler() transport.InboundMessageHandler { //nolint:gocyclo
+	connectionStore, err := did.NewConnectionStore(p)
+	if err != nil {
+		return func(*transport.Envelope) error { return err }
+	}
+
```

Comment:
as discussed, even though this function is called at the start of the handler only, it is better (more correct) to reference an existing connection store instead of creating a new one here.